client/src/components/RIC-Widget/ProductCard.jsx

```diff
@@ -4,14 +4,15 @@ import { auth } from '../../../../config.js';
 import AvgRating from '../AvgRating.jsx';
 import ActionButton from './ActionButton.jsx';
 
-function ProductCard({ productId, index, listState, triggerDelete }) {
+const ProductCard = ({ productId, index, listState, triggerDelete, triggerModal }) => {
   const [image, setImage] = useState('');
   const [category, setCategory] = useState('');
   const [name, setName] = useState('');
   const [price, setPrice] = useState(0);
-  const [salePrice, setSale] = useState(0);
+  const [salePrice, setSaleprice] = useState(0);
```

Comment:
Could combine the two of these into an object of the two prices instead of two hooks.
{
default: price,
sale:price
}